src/Module/Profile/Photos/Index.php

```diff
@@ -52,16 +61,265 @@ class Index extends \Friendica\Module\BaseProfile
 	private $app;
 	/** @var Database */
 	private $database;
+	/** @var SystemMessages */
+	private $systemMessages;
+	/** @var ACLFormatter */
+	private $aclFormatter;
 
-	public function __construct(Database $database, App $app, IManageConfigValues $config, App\Page $page, IHandleUserSessions $session, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	public function __construct(ACLFormatter $aclFormatter, SystemMessages $systemMessages, Database $database, App $app, IManageConfigValues $config, App\Page $page, IHandleUserSessions $session, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
 	{
 		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
 
-		$this->session  = $session;
-		$this->page     = $page;
-		$this->config   = $config;
-		$this->app      = $app;
-		$this->database = $database;
+		$this->session        = $session;
+		$this->page           = $page;
+		$this->config         = $config;
+		$this->app            = $app;
+		$this->database       = $database;
+		$this->systemMessages = $systemMessages;
+		$this->aclFormatter   = $aclFormatter;
+	}
+
+	protected function post(array $request = [])
+	{
+		$user = User::getByNickname($this->parameters['nickname'] ?? '');
+		if (!$user) {
+			throw new HTTPException\NotFoundException($this->t('User not found.'));
+		}
+
+		$page_owner_uid = intval($user['uid']);
+		$visitor        = $this->session->getLocalUserId() ? 0 : $this->session->getRemoteContactID($page_owner_uid);
```

Comment:
Using the `$visitor` doesn't make sense when `$can_post` will be `false` each time the `$visitor` is non-zero.

Reply:
$visitor is provided when creating the Photo recors later.

Reply:
Since we only continue with a local user, the `$visitor` will now always be `0`. So we don't need to set it any more, don't we?

Reply:
Ah, right, I missed the condition. I'll remove the superfluous code.

Reply:
Soon I'm of for the day. I guess I can have a short glance at it tomorrow.